sdk/runner/sdk.go

```diff
@@ -92,7 +98,14 @@ func (s *SDK) Create(req *api.CreateRunnerRequest, accountName, accountPassword
 		defer onError()
 		return nil, err
 	}
-	return s.Get(tx.Data)
+	run, err := s.Get(tx.Data)
+	if err != nil {
+		return nil, err
+	}
+	if !run.Hash.Equal(expRunnerHash) {
+		return nil, errors.New("calculated runner hash is not the same")
+	}
```

Comment:
Why we have this check here? We are getting runner from our cosmos rest backend - I don't see why this check is here.

Also why we get the runner in the first place when we already have it defined above.

Reply:
It's just to make sure that the transaction created the same hash.
The SDK is done by the client alone as opposed as the backend that is executed by all the node of the network (and more important the validators)

Reply:
But if the tx was already created with wrong hash shoudn't we take some action to remove it? Otherwise I'm not sure we should put it here

Reply:
> But if the tx was already created with wrong hash shoudn't we take some action to remove it? Otherwise I'm not sure we should put it here

good catch. it now call `onError` function that will stop the runner if there is an issue